extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -236,12 +236,26 @@ class LifecycleTask extends BackbeatTask {
                 }
             } else {
                 // 2. by VersionId, lower number means newer
+                //    NOTE: VersionId may be null
+                const doesVersionIdExist = (versions[vIdx].VersionId !== 'null'
+                    && deleteMarkers[dmIdx].VersionId !== 'null');
                 const isVersionVidNewer = (versions[vIdx].VersionId <
                     deleteMarkers[dmIdx].VersionId);
-                if (isVersionVidNewer) {
+                if (doesVersionIdExist && isVersionVidNewer) {
                     sortedList.push(versions[vIdx++]);
-                } else {
+                } else if (doesVersionIdExist) {
+                    sortedList.push(deleteMarkers[dmIdx++]);
+                } else if (versions[vIdx].LastModified >
```

Comment:
I'm not 100% sure we can use lexicographic comparison for dates, as it obviously depends on the format. I would feel safer to convert them to `Date` objects first, or add a comment stating it's on purpose and safe to do so.